src/functionalTest/java/uk/gov/hmcts/reform/professionalapi/FindUsersStatusByEmailTest.java

```diff
@@ -93,20 +104,19 @@ public void ac2_find_user_status_by_email_with_pui_case_manager_role_should_retu
         // creating user in idam with the same email used in the invite user
         // so that status automatically will be update in the up
         String email = userCreationRequest.getEmail();
-        RequestSpecification specification = professionalApiClient
+        RequestSpecification bearerToken = professionalApiClient
                 .getMultipleAuthHeadersExternal(puiUserManager,
-                userCreationRequest.getFirstName(),
-                userCreationRequest.getLastName(), userCreationRequest.getEmail());
+                        userCreationRequest.getFirstName(),

```

Comment:
i think you can optimise following call by storing bearertoken for pui-user-mnager or pui-case-manager. This will optimise sidam user creation and get berarer token call.

 RequestSpecification bearerToken = professionalApiClient
                .getMultipleAuthHeadersExternal(puiUserManager,
                        userCreationRequest.getFirstName(),
                        userCreationRequest.getLastName(), userCreationRequest.getEmail());

Reply:
This also applies for other classes

Reply:
This has been done here: https://github.com/hmcts/rd-professional-api/pull/857/commits/25db46d4a8fde2508d2d86558280ce9335aa96d9 and applied to all test classes where applicable